### ISSUE TITLE
Regression: MPI_Start() and MPI_PROC_NULL

### ISSUE THREAD

**Issue by dalcinl** [pmodels/mpich#5117]
This is MPICH 3.4.1 with `ch4:ucx`. I got a regression in mpi4py's testsuite.  Below a trivial reproducer. Same problem for `{S|B|R}send_init()`.  Additionally, `Recv_init()` deadlocks at `req.Wait()`.

```text
n [1]: from mpi4py import MPI                                                                                                                                  

In [2]: req = MPI.COMM_SELF.Send_init([MPI.BOTTOM, 0, MPI.BYTE], MPI.PROC_NULL, 0)                                                                              

In [3]: req.Start(); req.Wait(); req.Free()                                                                                                                     
[kw60439:671814:0:671814] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x36c6c6f63)
==== backtrace (tid: 671814) ====
 0 0x0000000000057fd3 ucs_debug_print_backtrace()  /home/dalcinl/Devel/BUILD/mpich-3.4.1/modules/ucx/src/ucs/debug/debug.c:656
 1 0x00000000000141e0 .annobin_sigaction.c()  sigaction.c:0
 2 0x00000000000524f1 ucp_tag_send_nbx()  /home/dalcinl/Devel/BUILD/mpich-3.4.1/modules/ucx/src/ucp/tag/tag_send.c:247
 3 0x0000000000053830 ucp_tag_send_nb()  /home/dalcinl/Devel/BUILD/mpich-3.4.1/modules/ucx/src/ucp/tag/tag_send.c:202
 4 0x0000000000287b75 MPIDI_UCX_send()  /home/dalcinl/Devel/BUILD/mpich-3.4.1/./src/mpid/ch4/netmod/include/../ucx/ucx_send.h:74
 5 0x000000000028a803 MPIDI_NM_mpi_isend()  /home/dalcinl/Devel/BUILD/mpich-3.4.1/./src/mpid/ch4/netmod/include/../ucx/ucx_send.h:224
 6 0x000000000028a803 MPIDI_isend_safe()  /home/dalcinl/Devel/BUILD/mpich-3.4.1/./src/mpid/ch4/src/ch4_send.h:334
 7 0x000000000028a803 MPID_Isend()  /home/dalcinl/Devel/BUILD/mpich-3.4.1/./src/mpid/ch4/src/ch4_send.h:516
 8 0x000000000028a803 MPID_Startall()  /home/dalcinl/Devel/BUILD/mpich-3.4.1/./src/mpid/ch4/src/ch4_startall.h:35
 9 0x000000000028a803 PMPI_Start()  /home/dalcinl/Devel/BUILD/mpich-3.4.1/src/mpi/request/start.c:86
```

**Comment by hzhou** [comment]
Thanks for reporting the bug. I believe this bug is fixed by https://github.com/pmodels/mpich/pull/5074.

**Comment by dalcinl** [comment]
@hzhou Indeed, #5074 is the proper fix. Sorry for the noise.

**Comment by raffenet** [comment]
I'll include the fix in a 3.4.2 release sometime soon.
